src/torchcodec/_core/SingleStreamDecoder.cpp

```diff
@@ -1171,27 +1178,33 @@ void SingleStreamDecoder::convertAudioAVFrameToFrameOutputOnCPU(
   int desiredSampleRate =
       streamInfo.audioStreamOptions.sampleRate.value_or(sourceSampleRate);
 
+  int sourceNumChannels = getNumChannels(srcAVFrame);
```

Comment:
So the "original" channels are actually a property of the frame, which means it's *likely* the same in a stream. I don't think we need to worry about it now, but: any chance there are a variable number of channels in a stream?

Reply:
Conceptually, it's a property of the samples rather than a property of the frame (which is what FFmpeg exposes).

I cannot guarantee that it doesn't exist, but I think it's very unlikely: samples are continuous, unlike video frames which are more discrete. So having samples with different properties in a single stream would be the result of a major bug in the encoder (which I guess we can't always rule out).

I'll add a `TORCH_CHECK(getNumChannels(srcAVFrame) == getNumChannels(avCodecContext));` to make sure we don't get any surprise